kaos/api/validators.py

```diff
@@ -0,0 +1,49 @@
+"""This module defines the validation decorators used by the various view functions."""
+
+from functools import wraps
+
+from flask import request
+
+from .errors import APIError, InputSchemaError
+
+
+def validate_request_schema(schema_validator):
+    """Validation decorator used to ensure that incomming requests to the view follow a specific
+    json schema.
+
+    Args:
+        schema_validator (fn): A scgema validation object used to check the request for errors.
+
+    Returns: A validation function decorator that wraps view functions in schema_validator.
+    """
+    def validation_fucntion(view_fn):
+        """Wraps a view function and ensures that the request matches the schema_validator.
+
+        Args:
+            view_fn (fn): The view function to protect.
+
+        Returns: The protected view function.
+        """
+
+        @wraps(view_fn)
```

Comment:
Ignore me if this sound stupid. Do you want to wrap the "def validation_fucntion()" instead of "def validated_function()", since everything inside "def validate_request_schema()" should be protected. If you keep this way, would the docstring of "validate_request_schema()" overwrite the docstring of "get_satellite_visibility()"?

Reply:
Right so to understand this you have to understand the way python decorators work:
A decorator is a function that produces another function ```@my_dec```:
```def my_dec(fn)```

A decorator that takes in a parameter ```@my_dec("wow")``` is a function that takes in the parameter to produce a function that takes in the decorated function to produce the wrapped function:
```
def my_dec(arg):
    def _intermediate(fn):
        def wrapped_function(...):
            ...
         return wrapped_function
    return _intermediate
```

So you see the wrapped_function is the function that eventually replaces the decorated function, we we need to call @wraps on the argument to that ```_intermediate``` function to preserve the docstring!

Reply:
I understand now. tyty